libpod/container_api.go

```diff
@@ -705,3 +705,23 @@ func (c *Container) Sync() error {
 
 	return nil
 }
+
+// RestartWithTimeout restarts a running container and takes a given timeout in uint
+func (c *Container) RestartWithTimeout(ctx context.Context, timeout uint) error {
+	if !c.batched {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return err
+		}
+	}
+
```

Comment:
If c.state.State is ContainerStateUnknown or ContainerStatePaused we should return an error

Reply:
fixed